ra/registration-authority.go

```diff
@@ -88,14 +88,21 @@ func validateEmail(address string, resolver core.DNSResolver) (rtt time.Duration
 	}
 	splitEmail := strings.SplitN(address, "@", -1)
 	domain := strings.ToLower(splitEmail[len(splitEmail)-1])
-	result, rtt, err := resolver.LookupHost(domain)
-	if err == nil && len(result) == 0 {
-		err = errEmptyDNSResponse
+	var rtt1, rtt2 time.Duration
+	var resultMX []string
+	var resultA []net.IP
+	resultMX, rtt1, err = resolver.LookupMX(domain)
+	if err == nil && len(resultMX) == 0 {
+		resultA, rtt2, err = resolver.LookupHost(domain)
+		if err == nil && len(resultA) == 0 {
+			err = errEmptyDNSResponse
+		}
 	}
 	if err != nil {
 		problem := dns.ProblemDetailsFromDNSError(err)
 		err = core.MalformedRequestError(problem.Detail)
 	}
+	rtt := rtt1 + rtt2
```

Comment:
Thanks for looking at this!

I haven't checked how the RTT is used; my guess was that it might be used for a timeout or something, so I thought it would be best to return the total elapsed time.

DNS queries can only ask for one type at a time, so you can't ask for MX and A and AAAA in one go. (ANY queries are magical and do not work the way you think they do.)